File: articles/the-future-of-algorithms-computational-reasoning/cpp/automation_risk.cpp
#include <algorithm>
#include <iostream>

double automation_risk(double stakes, double opacity, double delegation, double irreversibility) {
    return std::clamp(stakes * opacity * delegation * irreversibility, 0.0, 1.0);
}

int main() {
    std::cout << automation_risk(0.95, 0.85, 0.90, 0.80) << "\n";
}

File: articles/the-future-of-algorithms-computational-reasoning/rust/automation_risk.rs
fn automation_risk(stakes: f64, opacity: f64, delegation: f64, irreversibility: f64) -> f64 {
    (stakes * opacity * delegation * irreversibility).clamp(0.0, 1.0)
}

fn main() {
    println!("{:.6}", automation_risk(0.95, 0.85, 0.90, 0.80));
}
